extensions/wikia/UserLogin/FacebookSignupController.class.php

```diff
@@ -279,27 +301,25 @@ protected function getValidFbUserId() {
 	 * @return null|User
 	 */
 	protected function getValidWikiaUser( $wikiaUserName, $wikiaPassword ) {
+
 		if ( !$wikiaUserName ) {
-			$messageCode = 'userlogin-error-noname';
-			$errorParam = self::SIGNUP_USERNAME_KEY;
-		} else if ( !$wikiaPassword ) {
-			$messageCode = 'userlogin-error-wrongpasswordempty';
-			$errorParam = self::SIGNUP_PASSWORD_KEY;
-		} else {
-			$user = \User::newFromName( $wikiaUserName );
-			if ( !$user ) {
-				$messageCode = 'userlogin-error-nosuchuser';
-				$errorParam = self::SIGNUP_USERNAME_KEY;
-			} else {
-				if ( !$user->checkPassword( $wikiaPassword ) ) {
-					$messageCode = 'userlogin-error-wrongpassword';
-					$errorParam = self::SIGNUP_PASSWORD_KEY;
-				}
-			}
+			$this->setErrorResponse( 'userlogin-error-noname', [], self::SIGNUP_USERNAME_KEY);
+			return null;
+		}
+
+		if ( !$wikiaPassword ) {
+			$this->setErrorResponse( 'userlogin-error-wrongpasswordempty', [], self::SIGNUP_PASSWORD_KEY );
+			return null;
+		}
+
+		$user = \User::newFromName( $wikiaUserName );
+		if ( empty( $user ) ) {
```

Comment:
Nothing more than its sort of an unwritten standard at Wikia to use `empty()` to check for unset rather than applying logic like this directly, at least with a single variable.  Also, I like to test for positives where possible (empty is positive in the way that it asserts something rather than asserting not-something).